src/mux/tcp.rs

```diff
@@ -72,9 +73,15 @@ impl Channel {
 
             {
                 let mut shared = shared.borrow_mut();
-                /* TODO: Consider using async operations and select! */
```

Comment:
This looks incorrect. We have to let all protocols communicate, not just the one that we are waiting for.